api/grant.go

```diff
@@ -11,19 +11,22 @@ type Grant struct {
 	CreatedBy uid.ID `json:"created_by" note:"id of the identity that created the grant"`
 	Updated   Time   `json:"updated"`
 
-	Subject   uid.PolymorphicID `json:"subject" note:"a polymorphic field expecting an identity or group ID"`
-	Privilege string            `json:"privilege" note:"a role or permission"`
-	Resource  string            `json:"resource" note:"a resource name in Infra's Universal Resource Notation"`
+	Identity  uid.ID `json:"identity,omitempty"`
+	Group     uid.ID `json:"group,omitempty"`
+	Privilege string `json:"privilege" note:"a role or permission"`
+	Resource  string `json:"resource" note:"a resource name in Infra's Universal Resource Notation"`
 }
 
 type ListGrantsRequest struct {
-	Subject   uid.PolymorphicID `form:"subject"`
-	Resource  string            `form:"resource" example:"kubernetes.production"`
-	Privilege string            `form:"privilege" example:"view"`
+	Identity  uid.ID `json:"identity"`
+	Group     uid.ID `json:"group"`
```

Comment:
Oops, this comment didn't post for some reason.

I think we need the same `required_without` validation here. A query can't include both ofthese.

Reply:
This validation is slightly different since neither are required. I think the semantics you're looking for is that one of Identity or Group _can_ be specified but not both. From what I see in the validator docs, that's not an option so it will require custom validator. I'd suggest punting that to a separate issue

Reply:
Ahh yes, that's right, it should be "at most one of".

Does `excluded_if` not work for this?

Reply:
Ah yes that should work. I misread the description and thought it meant something else